src/Model.php

```diff
@@ -1633,15 +1637,15 @@ public function leftJoin($foreign_table, $defaults = [])
     protected function _hasReference($c, $link, $defaults = [])
     {
         if (!is_array($defaults)) {
-            $defaults = ['model' => $defaults ?: 'Model_'.$link];
+            $defaults = ['model' => $defaults ?: 'Model\\'.$link];
```

Comment:
so we still have `Model` prefix or namespace hard-coded?

Reply:
that's the rare case when you use:

```
$this->hasMany('User');
```

do you use that anywhere?

Reply:
Sure use it like that and a lot :)

Reply:
reverted.